### PR TITLE
Custom Node containing IronPython nodes shows dialog alert once per session

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

Do not let IronPython dialog alerts show more than once for a custom node in a session.
Used Custom Node ID to identify previously opened nodes.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [] All tests pass using the self-service CI.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@zeusongit looks like there are some test failures now.
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/1131/

File: src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void Loaded(ViewLoadedParams p)
         private void DisplayIronPythonDialog()
         {
             // we only want to create the dialog ones for each graph per Dynamo session, if the global setting is not disabled
-            if (DialogTracker.ContainsKey(CurrentWorkspace.Guid) || DynamoViewModel.IsIronPythonDialogDisabled)
+            if (DynamoViewModel.IsIronPythonDialogDisabled || DialogTracker.ContainsKey(CurrentWorkspace.Guid) || DialogTracker.ContainsKey((CurrentWorkspace as CustomNodeWorkspaceModel).CustomNodeId))
```
Comment (mjkkirschner) [inline src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:76]: can you make these lines shorter.

File: src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs [hunk 2]
```diff
@@ -87,6 +87,9 @@ private void DisplayIronPythonDialog()
             }), DispatcherPriority.Background);
 
             DialogTracker[CurrentWorkspace.Guid] = dialog;
+            if (CurrentWorkspace.GetType() == typeof(CustomNodeWorkspaceModel)){
```
Comment (mjkkirschner) [inline src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs:90]: you can just do `is`

File: test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs [hunk 3]
```diff
@@ -334,6 +334,47 @@ public void CustomNodeContainsIronPythonDependencyTest()
...
         }
 
+        /// <summary>
+        /// This test verifies that the IronPython dialog wont show the second time a custom node is opened
+        /// even if it contains IronPython nodes
+        /// </summary>
+        [Test]
+        public void WillNotDisplayDialogWhenOpeningCustomNodeWithIronPythonNodesSecondTimeInSameSession()
+        {
+            DebugModes.LoadDebugModesStatusFromConfig(Path.Combine(GetTestDirectory(ExecutingDirectory), "DynamoCoreWpfTests", "python2ObsoleteMode.config"));
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: maybe your test should disable this config mode before the next test?

Comment (zeusongit) [reply test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: I guess each test start its own instance of the sandbox so this should not be a problem, also, it is not disabled for any other test, before it as well which are passing.

Comment (mjkkirschner) [reply test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: Each test does not start it's own instance of sandbox - they run in two different scenarios.

on master-15 all tests run in a single process.
on master-5 there are some number of parallel processes with multiple tests inside, which tests is not deterministic.

Comment (zeusongit) [reply test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: ok, got it.

Comment (zeusongit) [reply test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: The problem was with the casting of CurrentWS into CustomNodeWS. Still have to figure out the other 2 failing tests, looking into those.

Comment (zeusongit) [reply test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs:344]: All test pass now

**Pull request merged**